src/google/google-calendars/googlecalendarsyncadaptor.cpp

```diff
@@ -99,11 +101,45 @@ void setGhostEventCleanupPerformed()
 
 QString gCalEventId(KCalCore::Incidence::Ptr event)
 {
-    return event->customProperty("jolla-sociald", "gcal-id");
+    // we abuse the comments field to store our gcal-id.
+    // we should use a custom property, but those are deleted on incidence deletion.
```

Comment:
RFC specifies COMMENT field intended for end user, but as calendar doesn't use it, nor google calendar synchronization, the abuse is borderline acceptable. Potential problem comes if event gets moved into another notebook where synchronization passes the value forward.